core/iwasm/common/wasm_c_api.c

```diff
@@ -2669,7 +2679,22 @@ wasm_func_call(const wasm_func_t *func, const wasm_val_vec_t *params,
     WASMExecEnv *exec_env = NULL;
     size_t param_count, result_count, alloc_count;
 
-    bh_assert(func && func->type && func->inst_comm_rt);
+    if (!func) {
+        return NULL;
+    }
+
+    if (!func->inst_comm_rt) {
+        wasm_name_t message = { 0 };
+        wasm_trap_t *trap;
+
+        wasm_name_new_from_string(&message, "the function isn't linked");
```

Comment:
message is not very friendly, had better be similiar with interpreter/aot, how about changing to `failed to call unlinked function`?

Reply:
indeed.